source/common/http/filter_manager.h

```diff
@@ -664,22 +664,13 @@ class FilterManager : public ScopeTrackedObject,
   void applyFilterFactoryCb(FilterContext context, FilterFactoryCb& factory) override;
 
   void log(AccessLog::AccessLogType access_log_type) {
-    RequestHeaderMap* request_headers = nullptr;
-    if (filter_manager_callbacks_.requestHeaders()) {
-      request_headers = filter_manager_callbacks_.requestHeaders().ptr();
-    }
-    ResponseHeaderMap* response_headers = nullptr;
-    if (filter_manager_callbacks_.responseHeaders()) {
-      response_headers = filter_manager_callbacks_.responseHeaders().ptr();
-    }
-    ResponseTrailerMap* response_trailers = nullptr;
-    if (filter_manager_callbacks_.responseTrailers()) {
-      response_trailers = filter_manager_callbacks_.responseTrailers().ptr();
-    }
+    Formatter::HttpFormatterContext log_context{filter_manager_callbacks_.requestHeaders().ptr(),
+                                                filter_manager_callbacks_.responseHeaders().ptr(),
+                                                filter_manager_callbacks_.responseTrailers().ptr()};
+    log_context.setAccessLogType(access_log_type);
```

Comment:
Can we add a 4-arg constructor so we can eliminate all these calls to `setAccessLogType()` and instead just pass it in to the constructor.

Reply:
In fact, we have a 5-arg constructor, let's use it directly.

Reply:
done.